examples/jsm/utils/BufferGeometryUtils.js

```diff
@@ -199,7 +202,25 @@ function mergeBufferGeometries( geometries, useGroups = false ) {
 		mergedGeometry.userData.mergedUserData = mergedGeometry.userData.mergedUserData || [];
 		mergedGeometry.userData.mergedUserData.push( geometry.userData );
 
-		if ( useGroups ) {
+		if ( useGroups && geometry.groups.length > 0 ) {
+
+			// Keep existing groups
+
+			let nextOffset = offset;
+
+			for ( let group of geometry.groups ) {
+
+				let offsetMaterialIndex = groupIndex + group.materialIndex;
+				mergedGeometry.addGroup( offset + group.start, group.count, offsetMaterialIndex );
+				nextOffset = Math.max( nextOffset, offset + group.start + group.count );
```

Comment:
It seems that "nextOffset" is set to the end of the final group, correct, and not the end of the full geometry? So the geometry will effectively be trimmed only on the end?

It might be a corner case but I don't think that's what I would expect. If geometry unreferenced by groups is going to be culled it should be done consistently (ie at the beginning and in the middle of the group ranges, as well). I think I'd expect all the geometry (vertices, indices, etc) to be merged the same way always with the groups being the only thing that's handled differently.

Reply:
Agreed. That would be unexpected.

I've pushed an additional change which addresses this issue. At this point, I've considered these cases; am I missing any?
- Contiguous groups
- Incrementing materialIndex
- Non-incrementing materialIndex
- Overlapping group start/ends
- Gaps between group start/ends
- Groups that don't span the entire position attribute
- Groups that don't start at 0

Here are cases that I'm intentionally not dealing with. I expect these are "invalid" groups:
- group.start < 0
- group.count < 0
- group.materialIndex < 0
- Groups where `group.start + group.count >= count` (Group reaches past the end of the geometry's vertices)